.github/workflows/golangci_lint.yml

```diff
@@ -17,17 +17,22 @@ jobs:
       - name: Install golangci-lint
         run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.54.2
       
-      # go.work makes it necessary to run linter manually
       - name: Run golangci-lint
-        run: find . -name "go.mod" -execdir $(go env GOPATH)/bin/golangci-lint run --enable=gofmt --tests=false --exclude-use-default --timeout=5m0s --out-format checkstyle:golangci-lint-report.xml \;
-      
-      - name: Check golangci-lint report for errors
-        run: find . -name "golangci-lint-report.xml" -exec grep "error" {} + && exit 1 || true
+        run: docker run --rm -v ${{ github.workspace }}:/app -w /app golangci/golangci-lint:v1.54.2 golangci-lint run --max-issues-per-linter 0 --max-same-issues 0 --out-format checkstyle:golangci-lint-results.xml,json:golangci-lint-results.json || true
+
+      - name: Check golangci-lint JSON report for errors
+        run: | 
+          if jq -e '.Issues | length > 0' golangci-lint-results.json; then
+              echo "Lint issues found:" 
+              jq '.Issues[] | "\(.FromLinter): \(.Pos.Filename):\(.Pos.Line) \(.Text)"' golangci-lint-results.json
+              exit 1
+          else 
+              echo "No issues found"
+          fi
 
-      - name: Upload golangci-lint report
+      - name: Upload golangci-lint JSON report
```

Comment:
You still upload xml and download it as XML [to sonar](https://github.com/smartcontractkit/chainlink-common/actions/runs/6896750212/job/18763609935?pr=246#step:7:3), I am not sure it will be correct to tie it to JSON

Reply:
I'll fix the typo, my bad. The artifact that is uploaded is in xml format.